app/addons/auth/actions.js

```diff
@@ -10,148 +10,157 @@
+    return errorHandler({message: app.i18n.en_US['auth-passwords-not-matching']});
+  }
   var nodes = nodesStore.getNodes();
-  var promise = FauxtonAPI.session.createAdmin(username, password, loginAfter, nodes[0].node);
+  Api.createAdmin({
+    name: username,
+    password,
+    node: nodes[0].node
+  }).then(
+    () => {
+      dispatch(passwordUpdated());
+    },
+    errorHandler
+  );
+};
 
-  promise.then(() => {
-    FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.adminCreated });
-    if (loginAfter) {
-      FauxtonAPI.navigate('/');
-    } else {
-      FauxtonAPI.dispatch({ type: ActionTypes.AUTH_CLEAR_CREATE_ADMIN_FIELDS });
-    }
-  }, (xhr, type, msg) => {
-    msg = xhr;
-    if (arguments.length === 3) {
-      msg = xhr.responseJSON.reason;
+export const adminCreated = () => {
+  return {
+    type: AUTH_ADMIN_CREATED
+  };
+};
+
+export const createAdmin = (username, password, loginAfter) => (dispatch) => {
+  const node = nodesStore.getNodes()[0].node;
+  if (!validateUser(username, password)) {
+    return errorHandler({message: app.i18n.en_US['auth-missing-credentials']});
+  }
+
+  Api.createAdmin({name: username, password, node})
+  .then(resp => {
+    if (resp.error) {
+      return errorHandler({message: `${app.i18n.en_US['auth-admin-creation-failed-prefix']} ${resp.reason}`});
     }
-    errorHandler(FauxtonAPI.session.messages.adminCreationFailedPrefix + ' ' + msg);
-  });
-}
 
-// simple authentication method - does nothing other than check creds
-function authenticate (username, password, onSuccess) {
-  $.ajax({
-    cache: false,
-    type: 'POST',
-    url: '/_session',
-    dataType: 'json',
-    data: { name: username, password: password }
-  }).then(() => {
-    FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_CREDS_VALID,
-      options: { username: username, password: password }
-    });
-    hidePasswordModal();
-    onSuccess(username, password);
-  }, () => {
     FauxtonAPI.addNotification({
-      msg: 'Your password is incorrect.',
-      type: 'error',
-      clear: true
+      msg: app.i18n.en_US['auth-admin-created']
     });
-    FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_CREDS_INVALID,
-      options: { username: username, password: password }
-    });
-  });
-}
-
-function updateCreateAdminUsername (value) {
-  FauxtonAPI.dispatch({
-    type: ActionTypes.AUTH_UPDATE_CREATE_ADMIN_USERNAME_FIELD,
-    value: value
-  });
-}
 
-function updateCreateAdminPassword (value) {
-  FauxtonAPI.dispatch({
-    type: ActionTypes.AUTH_UPDATE_CREATE_ADMIN_PWD_FIELD,
-    value: value
+    dispatch(adminCreated());
+    if (loginAfter) {
+      return FauxtonAPI.navigate("/login");
+    }
   });
-}
+};
 
-function selectPage (page) {
-  FauxtonAPI.dispatch({
-    type: ActionTypes.AUTH_SELECT_PAGE,
-    page: page
-  });
+// simple authentication method - does nothing other than check creds
+export function authenticate(username, password, onSuccess) {
```

Comment:
Consider being consistent with arrow functions here.  I'm perhaps overly picky when it comes to consistency :)